Core/include/Acts/TrackFitting/detail/GainMatrixUpdaterImpl.hpp

```diff
@@ -59,35 +69,32 @@ std::tuple<double, std::error_code> GainMatrixUpdater::visitMeasurementImpl(
     return {0, KalmanFitterError::UpdateFailed};
   }
 
-  trackState.filtered =
-      trackState.predicted + K * (calibrated - H * trackState.predicted);
+  filteredParameters =
+      predictedParameters + K * (calibrated - H * predictedParameters);
   // Normalize phi and theta
-  trackState.filtered = normalizeBoundParameters(trackState.filtered);
-  trackState.filteredCovariance =
-      (BoundSquareMatrix::Identity() - K * H) * trackState.predictedCovariance;
-  ACTS_VERBOSE("Filtered parameters: " << trackState.filtered.transpose());
-  ACTS_VERBOSE("Filtered covariance:\n" << trackState.filteredCovariance);
-
-  ParametersVector residual;
-  residual = calibrated - H * trackState.filtered;
-  ACTS_VERBOSE("Residual: " << residual.transpose());
+  filteredParameters = normalizeBoundParameters(filteredParameters);
+  filteredCovariance =
+      (BoundSquareMatrix::Identity() - K * H) * predictedCovariance;
+  ACTS_VERBOSE("Filtered parameters: " << filteredParameters.transpose());
+  ACTS_VERBOSE("Filtered covariance:\n" << filteredCovariance);
 
-  CovarianceMatrix m =
-      ((CovarianceMatrix::Identity() - H * K) * calibratedCovariance).eval();
```

Comment:
Why is this wrong? As far as I can tell this is exactly what's described by Frühwirth:

$R_k = (I - H_kK_k)V_k$